modules/programs/alacritty/default.nix

```diff
@@ -0,0 +1,234 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  defaults = import ./default-config.nix;
+  cfg = config.programs.alacritty;
+
+  mapAttrNamesRecursive' = f: set:
+    let
+      recurse = set:
+        let
+          g =
+            name: value:
+            if isAttrs value
+              then nameValuePair (f name) (recurse value)
+              else nameValuePair (f name) value;
+        in mapAttrs' g set;
+    in recurse set;
+
+  toSnakeCase = replaceChars upperChars (map (c: "_${c}") lowerChars);
+
+  animationList = [
+    "Ease"
+    "EaseOut"
+    "EaseOutSine"
+    "EaseOutQuad"
+    "EaseOutCubic"
+    "EaseOutQuart"
+    "EaseOutQuint"
+    "EaseOutExpo"
+    "EaseOutCirc"
+    "Linear"
+  ];
+  actionList = [
+    "Paste"
+    "PasteSelection"
+    "Copy"
+    "IncreaseFontSize"
+    "DecreaseFontSize"
+    "ResetFontSize"
+    "ScrollPageUp"
+    "ScrollPageDown"
+    "ScrollToTop"
+    "ScrollToBottom"
+    "ClearHistory"
+    "Hide"
+    "Quit"
+    "ClearLogNotice"
+  ];
+  modList = [
+    "Command"
+    "Control"
+    "Shift"
+    "Alt"
+  ];
+  modeList = [
+    "~AppCursor"
+    "AppCursor"
+    "~AppKeypad"
+    "AppKeypad"
+  ];
+in {
+  options.programs.alacritty = with types; {
+    enable = mkEnableOption "Alacritty terminal";
+    window = {
+      dimensions = {
+        columns = mkOption {
+          type = int;
+          default = 80;
+          description = "Number of columns in the window";
+        };
+        lines = mkOption {
+          type = int;
+          default = 24;
+          description = "Number of lines in the window";
+        };
+      };
+      padding = {
+        x = mkOption {
+          type = int;
+          default = 2;
+          description = "Horizontal padding around the window, in pixels";
+        };
+        y = mkOption {
+          type = int;
+          default = 2;
+          description = "Vertical padding around the window, in pixels";
+        };
+      };
+      decorations = mkOption {
+        description = ''
+          Window decorations
+            - full: Borders and title bar
+            - none: Neither borders nor title bar
+        '';
+        type = enum [ "none" "full" ];
+        default = "full";
+      };
+    };
+    scrolling = {
+      history = mkOption {
+        type = int;
+        description = ''
+          Maximum number of lines in the scrollback buffer.
+          Specifying '0' will disable scrolling.
+        '';
+        default = 10000;
+      };
+      multiplier = mkOption {
+        type = int;
+        description = "Number of lines the viewport will move for every line scrolled.";
+        default = 3;
+      };
+      autoScroll = mkOption {
+        type = bool;
+        description = "Scroll to the bottom when new text is written to the terminal.";
+        default = false;
+      };
+    };
+    font = let
+      fontOption = {
+        family = mkOption {
+          type = string;
```

Comment:
It is best to avoid the `string` type and instead use `str`. The `string` type behaves unexpectedly under merging while `str` doesn't allow merging at all:

```
nix-repl> lib.types.string.merge ["test"] [ {file="f";value="a";} {file="g";value="b";} ]
"ab"

nix-repl> lib.types.str.merge ["test"] [ {file="f";value="a";} {file="g";value="b";} ]
error: The unique option `test' is defined multiple times, in `f' and `g'.
```